File: JEJE/고정올-15975 화살표 그리기.py
answer = 0
tc = int(input()) #테스트 케이스 입력 받음
arr = [[]for i in range(tc+1)] # [[],[],[]] 만들기

for i in range(tc): #색깔별로 분류
    x, color = list(map(int, input().split()))
    arr[color].append(x)

for i in range(len(arr)): #합 구하기 시작
    if len(arr[i]) > 1: #만약 같은 색깔 상에서 좌표가 1개보다 많으면
        arr[i].sort() #정렬하고
        
        for j in range(len(arr[i])): #이제 좌표 하나씩 읽기 시작함
            if j==0: 
                answer += abs(arr[i][j]-arr[i][j+1]) #맨 처음 좌표
            elif j==len(arr[i])-1: 
                answer += abs(arr[i][j]-arr[i][j-1]) #맨 마지막 좌표
            else:
                answer += min(abs(arr[i][j]-arr[i][j+1]), abs(arr[i][j]-arr[i][j-1])) #사이에 있는 좌표들

#min(a,b): a와 b중 최솟값을 알려준다
#abs(): 절댓값을 리턴함(알려줌)

#a+=b: a = a+b
#추가로 a-=b: a = a-b
#*=, /=도 있음

print(answer) #정답 출력

File: JEJE/고정올-15975 화살표 그리기.cpp
// SPDX-License-Identifier: AGPL-3.0-or-later
// NOTICE: This code is restricted from being used for any AI or machine learning training purposes.
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main()
{
	ios_base::sync_with_stdio(NULL);
	cin.tie(NULL); cout.tie(NULL);
	
	long int tc, answer = 0;
	cin >> tc;
	vector<vector<long int>>  v(tc+1, vector <long int>(1, 0));
	for (int i = 0; i < tc; i++)
	{
		int x, color;
		cin >> x >> color;
		v[color].push_back(x);
	}
	
	for (int i = 0; i<v.size(); i++)
	{
		if (v[i].size() > 2)
		{
			sort(v[i].begin(),v[i].end());
			answer += v[i][2] - v[i][1];
			for (int j = 2; j < v[i].size()-1; j++)
			{
				answer += min(v[i][j + 1] - v[i][j], v[i][j] - v[i][j - 1]);
			}
			answer += v[i][v[i].size() - 1] - v[i][v[i].size() - 2];
		}
		
	}
	cout << answer << "\n";

}